### PR TITLE
Introduce the << operator to get Quantity views of arrays

### PR DESCRIPTION
**Pull request by mhvk**
This is a replacement of #7652, using `array << unit` instead of `array | unit` to get `Quantity` views of arrays, and to change units of quantities. It includes updates to the documentation (also the performance section). For discussion, see #7652.

### CODE REVIEW

File: astropy/units/quantity.py [hunk 1]
```diff
@@ -858,6 +858,59 @@ def __ne__(self, other):
...
+
+    def __rlshift__(self, other):
+        if not self.isscalar:
+            return NotImplemented
+        return Unit(self).__rlshift__(other)
+
+    # Give warning for other >> self, since probably other << self was meant.
+    def __rrshift__(self, other):
+        warnings.warn(">> is not implemented. Did you mean to convert "
+                      "something to this quantity as a unit using '>>'?",
```
Comment (astrofrog) [inline astropy/units/quantity.py]: Should this be ``using '<<'``?

File: astropy/units/tests/test_quantity.py [hunk 2]
```diff
@@ -260,6 +260,80 @@ class MyQuantityLookalike(np.ndarray):
         with pytest.raises(TypeError):
             u.Quantity(mylookalike)
 
+    def test_creation_via_view(self):
+        # This works but is no better than 1. * u.m
+        q1 = 1. << u.m
+        assert isinstance(q1, u.Quantity)
+        assert q1.unit == u.m
+        assert q1.value == 1.
+        # With an array, we get an acual view.
```
Comment (astrofrog) [inline astropy/units/tests/test_quantity.py]: acual -> actual

File: astropy/units/quantity.py (hunk 1 above)
Comment (mhvk) [reply astropy/units/quantity.py]: Oops... fixed.

File: astropy/units/tests/test_quantity.py (hunk 2 above)
Comment (mhvk) [reply astropy/units/tests/test_quantity.py]: thanks, fixed.

**Comment by taldcroft** [comment]
@astrofrog -

No, that should not modify the table value.  When you create a new column like `t['a'] = something`, it makes a copy.

In fact it occurs to me now that I'm not sure you can add a column to an existing table *without* making a copy.  Need to ponder.

BTW, you should be using QTable in this situation if you actually care about units!!  :smile:

**Comment by mhvk** [comment]
`<< unit` does make a copy if a unit conversion is actually needed.

**Comment by astrofrog** [comment]
@taldcroft - just to clarify, I know that ``t['a'] = [1,2,3] * u.m`` makes a copy but I was more curious about whether ``t['a'] << u.cm`` did. But I see @mhvk's point that a copy is expected here.

**Comment by astrofrog** [comment]
It seems there haven't been any objections to this. Would it make sense to just go ahead and merge?

**Comment by mhvk** [comment]
Sort-of forgot about this, but will merge now.

**Pull request merged**